pkg/sbproxy/sbproxy.go

```diff
@@ -0,0 +1,227 @@
+package sbproxy
+
+import (
+	"context"
+	"strconv"
+	"sync"
+
+	"os"
+	"os/signal"
+
+	"time"
+
+	"fmt"
+
+	"net/http"
+
+	"github.com/Peripli/service-broker-proxy/pkg/osb"
+	"github.com/Peripli/service-broker-proxy/pkg/platform"
+	"github.com/Peripli/service-broker-proxy/pkg/sbproxy/middleware"
+	"github.com/Peripli/service-broker-proxy/pkg/sm"
+	"github.com/Peripli/service-broker-proxy/pkg/task"
+	"github.com/gorilla/mux"
+	"github.com/pmorie/osb-broker-lib/pkg/metrics"
+	"github.com/pmorie/osb-broker-lib/pkg/rest"
+	"github.com/pmorie/osb-broker-lib/pkg/server"
+	prom "github.com/prometheus/client_golang/prometheus"
+	"github.com/robfig/cron"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	BrokerPathParam = "brokerID"
+	ApiPrefix       = "/v1/osb/{" + BrokerPathParam + "}"
+)
+
+var (
+	ctx    context.Context
+	cancel context.CancelFunc
+	group  sync.WaitGroup
+)
+
+type Configuration interface {
+	Validate() error
+	SbproxyConfig() *ServerConfiguration
+	OsbConfig() *osb.ClientConfiguration
+	SmConfig() *sm.ClientConfiguration
+	PlatformConfig() platform.ClientConfiguration
+}
+
+type SBProxy struct {
+	CronScheduler *cron.Cron
+	Server        *server.Server
+	ServerConfig  *ServerConfiguration
+}
+
+func New(config Configuration) (*SBProxy, error) {
+	if config == nil {
+		return nil, fmt.Errorf("config cannot be nil")
+	}
+	if err := config.Validate(); err != nil {
+		return nil, err
+	}
+
+	sbproxyConfig := config.SbproxyConfig()
+	setUpLogging(sbproxyConfig.LogLevel, sbproxyConfig.LogFormat)
+
+	osbServer, err := defaultOSBServer(config.OsbConfig())
+	if err != nil {
+		return nil, err
+	}
+
+	osbServer.Router.Use(middleware.LogRequest())
+
+	cronScheduler := cron.New()
+	regJob, err := defaultRegJob(config.PlatformConfig(), config.SmConfig(), config.SbproxyConfig().Host)
+	if err != nil {
+		return nil, err
+	}
+	cronScheduler.AddJob("@every 1m", regJob)
```

Comment:
you don't need a separate dependency for that
you can just use `time.NewTicker(time.Minute)`